.gitbook/5-oracles/1-operate-oracle-nodes/2-oracle-intialization.md

```diff
@@ -0,0 +1,144 @@
+enclave.
+
+```bash
+# If you run oracle on your host
+sudo mkdir /oracle
+
+# If you run oracle using Docker
+mkdir $(pwd)/oracle
+```
+
+After that, if you want to run the `oracled` with Docker as described in
+the [oracle-installation](./1-oracle-installation.md),
+it is recommended to create an environment variable that you can execute the Docker container easily.
+
+```bash
+export DOCKER_CMD="docker run --rm \
+  --device /dev/sgx_enclave \
+  --device /dev/sgx_provision \
+  -v $(pwd)/oracle:/oracle ghcr.io/medibloc/panacea-oracle:latest"
+```
+
+If you are not going to run the `oracled` without Docker, you can set an `DOCKER_CMD` environment variable empty string.
+
+```bash
+export DOCKER_CMD=""
+```
+
+## Command Line of Initialization
+
+```bash
+$DOCKER_CMD ego run oracled init --home $HOME/.oracle 
+```
+
+When run the above CLI for initializing the oracle, the `config.toml` file will be generated under the `$HOME/.oracle`
+in the enclave.
+The `config.toml` file will be shown like this:
+
+```toml
+# This is a TOML config file.
+# For more information, see https://github.com/toml-lang/toml
+
+###############################################################################
+###                           Base Configuration                            ###
+###############################################################################
+
+log-level = "info"
+oracle-acc-num = "0"
+oracle-acc-index = "0"
+data-dir = "data"
+
+oracle-priv-key-file = "oracle_priv_key.sealed"
+oracle-pub-key-file = "oracle_pub_key.json"
+node-priv-key-file = "node_priv_key.sealed"
+
+###############################################################################
+###                         Panacea Configuration                           ###
+###############################################################################
+
+[panacea]
+
+chain-id = ""
+grpc-addr = "http://127.0.0.1:9090"
+rpc-addr = "tcp://127.0.0.1:26657"
+default-gas-limit = "400000"
+default-fee-amount = "2000000umed"
+
+# A primary RPC address for light client verification
+
+light-client-primary-addr = "tcp://127.0.0.1:26657"
+
+# Witness addresses (comma-separated) for light client verification
+
+light-client-witness-addrs = "tcp://127.0.0.1:26657"
+
+# Setting log information for light client
+
+light-client-log-level = "error"
+
+###############################################################################
+###                         IPFS Configuration                           ###
+###############################################################################
+
+[ipfs]
+
+ipfs-node-addr = "127.0.0.1:5001"
+
+###############################################################################
+###                         API Configuration                           ###
+###############################################################################
+
+[api]
+
+listen-addr = "127.0.0.1:8080"
+write-timeout = "60"
+read-timeout = "15"
+```
+
```

Comment:
I think it would be nice if you could provide instructions on how to modify which field in `config.toml`, e.g. chain-id = "panacea-3".

Reply:
The operator of oracle have to change some configurations, for example mnemonic, chain-id, etc.
So, I meant how about providing explanation for those configurations (must to be modified), excepting those with default values.

Reply:
Ah, does it mean that the argument is required or not?

Reply:
As we discussed offline, the every component of `config.toml` will be not introduced in this document. As similar as [cosmos sdk docs](https://docs.cosmos.network/v0.46/run-node/run-node.html), the only component that can be modified or updated will be introduced. The other components will be explained using comment in `config.toml`.

Reply:
I reflected in [3a2beb8](https://github.com/medibloc/panacea-core/pull/611/commits/3a2beb86691e3eaaf09fe07bc4fe9b961716354c). Please review again @H4NLee .